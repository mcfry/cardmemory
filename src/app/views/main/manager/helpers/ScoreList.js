import React from "react";

class ScoreList extends React.PureComponent {
	render() {
		const { bestTimes } = this.props;
		let difficultyStr = '';
		let difficultyObj = null;

		if (bestTimes.easy !== undefined) {
			difficultyStr = 'Easy';
			difficultyObj = bestTimes.easy;
		} else if (bestTimes.medium !== undefined) {
			difficultyStr = 'Medium';
			difficultyObj = bestTimes.medium;
		} else if (bestTimes.hard !== undefined) {
			difficultyStr = 'Hard';
			difficultyObj = bestTimes.hard;
		}

		return (
			<div className="list-group" style={{'paddingBottom': '20px'}}>
				<div>
					<h1 style={{'marginLeft': '20px'}}>{difficultyStr}</h1>
				</div>
				{difficultyObj.map((time, index) => {
					return (
						<button key={index} className="list-group-item list-group-item-center">
						    <span className="scoreboard-col"><i className="fa fa-clock-o"></i>&nbsp;Seconds: {time.seconds}</span>
						    <span className="scoreboard-col"><i className="fa fa-times"></i>&nbsp;Mistakes: {time.mistakes}</span>
						</button>
					)
				})}
			</div>
		);
	}
}

export default ScoreList;