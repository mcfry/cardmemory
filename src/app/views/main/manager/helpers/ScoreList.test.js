import React from 'react';
import renderer from 'react-test-renderer';
import ScoreList from './ScoreList';

test('Changes when difficulty, seconds, and mistakes change', () => {
	const comp = renderer.create(
		<ScoreList bestTimes={{easy: [{seconds: 100, mistakes: 2}]}}/>
	);

	let tree = comp.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.bestTimes = {
		medium: [{seconds: 105, mistakes: 1}]
	};
	tree = comp.toJSON();
	expect(tree).toMatchSnapshot();
});