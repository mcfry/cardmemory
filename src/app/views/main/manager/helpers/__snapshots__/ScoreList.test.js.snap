// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changes when difficulty, seconds, and mistakes change 1`] = `
<div
  className="list-group"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <div>
    <h1
      style={
        Object {
          "marginLeft": "20px",
        }
      }
    >
      Easy
    </h1>
  </div>
  <a
    className="list-group-item list-group-item-center"
  >
    <span
      className="scoreboard-col"
    >
      <i
        className="fa fa-clock-o"
      />
       Seconds: 
      100
    </span>
    <span
      className="scoreboard-col"
    >
      <i
        className="fa fa-times"
      />
       Mistakes: 
      2
    </span>
  </a>
</div>
`;

exports[`Changes when difficulty, seconds, and mistakes change 2`] = `
<div
  className="list-group"
  style={
    Object {
      "paddingBottom": "20px",
    }
  }
>
  <div>
    <h1
      style={
        Object {
          "marginLeft": "20px",
        }
      }
    >
      Easy
    </h1>
  </div>
  <a
    className="list-group-item list-group-item-center"
  >
    <span
      className="scoreboard-col"
    >
      <i
        className="fa fa-clock-o"
      />
       Seconds: 
      100
    </span>
    <span
      className="scoreboard-col"
    >
      <i
        className="fa fa-times"
      />
       Mistakes: 
      2
    </span>
  </a>
</div>
`;
